### DISCUSSION TITLE
Seeking a Cross-Platform Secrets Management Solution

### DISCUSSION THREAD

**Discussion by anose001** [AvaloniaUI/Avalonia#16866] [Q&A]
Is it possible to implement .NET MAUI SecureStorage for Cross-Platform Secrets Management without setting up an AvaloniaMauiHybrid project? See this [link](https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/storage/secure-storage?view=net-maui-8.0&tabs=android) for reference.

Maybe there are other solutions out there?

Edit: 
For **mobile (Android, iOS)** I found Xamarin.Essentials  and the SecureStorage class therein
For **desktop (Windows, macOS, Linux)** I found [keychain](https://github.com/hrantzsch/keychain) via [KeySharp](https://github.com/goaaats/KeySharp?tab=readme-ov-file) but have not tested them yet

**Comment by thevortexcloud** [comment]
If you want Avalonia specifically, there is the official Github credential manager which can probably be used for some inspiration for making your own.

https://github.com/git-ecosystem/git-credential-manager

**Reply by anose001** [reply]
Oh nice, Thanks for the suggestion!

**Reply by dariostrm** [reply]
Hey bro, how did it go? Did you find something?

**Reply by bairog** [reply]
@thevortexcloud There is a .NET wrapper for GCM ([Devlooped.CredentialManager](https://github.com/devlooped/CredentialManager)). But like GCM itself it supports [only desktop](https://github.com/git-ecosystem/git-credential-manager/discussions/1739) (Windows, macOS, Linux). Do you have any ideas for mobile (Android, iOS)?

@anose001 Did you find any solution for both desktop and mobile?

**Reply by thevortexcloud** [reply]
When I said git credential manager, I did not mean to use it verbatim. Merely to just use it as inspiration for rolling your own. Android and iOS both have secret management and credential management APIs for various different things. But I can't say I have any experience using them.
